### PR TITLE
Add FFmpeg tagloader

### PR DESCRIPTION
**Pull request by notspiff**
Use this for .mka and .dsf.

Need project updates. Here you go @da-anda

### CODE REVIEW

**Comment by da-anda** [comment]
awesome @notspiff thanks.

ping @Memphiz @mkortstiege mind doing some Xcode monkey work? @ace20022 or @Karlson2k, could one of you add it to the windows solution?

**Comment by Memphiz** [comment]
when i am home yep

**Comment by Memphiz** [comment]
xcode sync:

https://github.com/Memphiz/xbmc/commit/f5a7218bc91b8d6ced125ca8c005363ad69ad356

**Comment by notspiff** [comment]
removed variable and some other cleanup, picked xcode sync.

**Comment by Montellese** [comment]
Thanks for extracting this from the audiobook PR.

**Comment by ace20022** [comment]
VS sync and a win32 "fix" off_t has the wrong type: https://github.com/ace20022/xbmc/commits/add_ffmpeg_tagloader

**Comment by notspiff** [comment]
rebased (squashed the fix), took care of @Karlson2k concern and picked visual studio sync.

File: xbmc/music/tags/MusicInfoTagLoaderFFmpeg.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+bool CMusicInfoTagLoaderFFmpeg::Load(const std::string& strFileName, CMusicInfoTag& tag, EmbeddedArt *art)
+{
+  tag.SetLoaded(false);
+
+  CFile file;
+  if (!file.Open(strFileName))
+    return false;
+
+  uint8_t* buffer = (uint8_t*)av_malloc(32768);
```
Comment (Montellese) [inline xbmc/music/tags/MusicInfoTagLoaderFFmpeg.cpp]: Where does the magic number come from? Might be worth putting it in a define or const as you use it multiple times.

Comment (notspiff) [inline xbmc/music/tags/MusicInfoTagLoaderFFmpeg.cpp]: magic number comes from ffmpeg. moved a define and reused it.

**Comment by ace20022** [comment]
Please, reconsider your decision.

**Comment by da-anda** [comment]
I know it's not the place for it, but yes, Arne, please reconsider. Your work is very much welcome. At least elaborate why you closed all your PRs - bikeshedding? If that's the case I don't see where it would have happened. I hope it was just some overreaction because of something that pissed you off and everything is cool again next morning.

**Comment by notspiff** [comment]
I warned up front i would not tolerate any nonsense. I asked up front, was told kiss was all good, put in the work to fix the bug and bam! Now we need fftw cause somebody read on some web page that it is faster. Classic pr bullshit, not sticking to the code at hand.

**Comment by da-anda** [comment]
well, the follow up comment by topfs was that fftw could maybe be considered at a later point but that your work should go in none the less to fix the bug. At least as I understood it nobody objected this. 

Ah, I just read https://github.com/notspiff/kodi-cmake/commit/960f8063815827bea419a605dd1c62b3c604ed78 which I didn't know about (I turned off github spam so I don't get any mail spam unless somebody pings me with @da-anda), just as probably most many team members didn't read that discussion. You can't be pissed only because one of our audio DSP guys mentioned fftw again and didn't know about what's been discussed with fritsch and Fernet. A simple "sorry, I won't work on fftw because of https://github.com/notspiff/kodi-cmake/commit/960f8063815827bea419a605dd1c62b3c604ed78 so take it or leave it" would have done as well IMO. At least to me it feels like a little over reaction on your end. So please cool down, drink a cool Dahls (that was your fav beer, right?) and reconsider :)

**Pull request closed without merging**